web/frontend/src/pages/form/components/utils/useQuestionForm.ts

```diff
@@ -32,31 +32,34 @@ const useQuestionForm = (initState: RankQuestion | SelectQuestion | TextQuestion
         case 'addChoiceRank':
           setState({
             ...state,
-            ChoicesMap: newChoicesMap,
+            ChoicesMap: { ChoicesMap: newChoicesMap, URLs: [...ChoicesMap.URLs, ''] },
             MaxN: Math.max(
-              ChoicesMap.get('en').length + 1,
-              ChoicesMap.get('fr').length + 1,
-              ChoicesMap.get('de').length + 1
+              ChoicesMap.ChoicesMap.get('en').length + 1,
```

Comment:
I propose to replace it with this (thanks to Claude):

```
Math.max(...ChoicesMap.ChoicesMap.values()) + 1
```

Which will even work if some of the languages are not present.